externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/util/ResultSetWrapper.scala

```diff
@@ -30,6 +30,7 @@ class ResultSetWrapper(statement: Statement)
   private lazy val metadata = currentResult.getMetaData
 
   override def hasNext: Boolean = {
+    if (currentResult.isClosed) return false
```

Comment:
`FetchResult` may be called again when the previous resultSet has ended, so we need to determine whether the resultSet has been closed.

https://github.com/apache/kyuubi/blob/f05aefb86643e69c64237ebfb9db208ec509f3b2/kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiQueryResultSet.java#L328